### PR TITLE
[TestGap] New tests to cover VNET route advertisement.

### PR DESCRIPTION
**Pull request by siqbal1986**
### Description of PR

Summary:
Fixes # (issue)
Added new tests to cover the previously untested scenario. These tests create VNET Vxlan ECMP routes and then chekc if they are advertised to the upstream neighbors.
### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [X] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### How did you do it?
Create the BGP profile 
Create a VNET routes.
check neighbor bgp routes to verify the advertisements.
The following tests are performed for Both V4 and V6 routes.

Step | Goal | Expected results
-- | -- | --
Create a tunnel route and advertise the tunnel route to all neighbor without community id | BGP | ALL BGP neighbors can recieve the advertised BGP routes
Create a tunnel route and advertise the tunnel route to all neighbor with community id | BGP | ALL BGP neighbors can recieve the advertised BGP routes with community id
Update a tunnel route and advertise the tunnel route to all neighbor with new community id | BGP | ALL BGP neighbors can recieve the advertised BGP routes with new community id
Create a tunnel route and advertise the tunnel route to all neighbor with BGP profile, but create the profile later | BGP | ALL BGP neighbors can recieve the advertised BGP routes without community id first, after the profile table created, the community id would be added and all BGP neighbors can recieve this update and associate the community id with the route
Delete a tunnel route | BGP | ALL BGP neighbors can remove the previously advertised BGP routes
Create 400 tunnel routes and advertise all tunnel routes to all neighbor with community id | BGP scale | ALL BGP neighbors can recieve 400 advertised BGP routes with community id and record the time
Updat BGP_PROFILE_TABLE with new community id for 400 tunnel routes and advertise all tunnel routes to all neighbor with new community id | BGP scale | ALL BGP neighbors can recieve 400 advertised BGP routes with new community id and record the time
#### How did you verify/test it?
image

#### Any platform specific information?
These scale tests are set to un with 400 routes. Altough I have ran these tests with 4k routes without any problem, but that takes the test run time to around 40 minutes. 
#### Supported testbed topology if it's a new test case?
T1 Cisco, T1 Mlnx, VS

### CODE REVIEW

File: tests/vxlan/test_vxlan_route_advertisement.py [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+
+    def verify_nighbor_doesnt_have_routes(self, routes, community=""):
+        if self.prefix_type == 'v4':
+            prefix_mask = 24
+        else:
+            prefix_mask = 64
+        for vnet in routes:
+            for prefix in routes[vnet]:
+                route = f'{prefix}/{prefix_mask}'
+                result = self.vxlan_test_setup['t2']['host'].get_route(route)
```
[the remark this answers is not in the document]
Comment (siqbal1986) [reply tests/vxlan/test_vxlan_route_advertisement.py:276]: if the advertisement is happening, then it either reach all T2s or none of them.  The get_route operation is slow and time consuming.  In order to speed up the test, i checked only one t2.

[the remark this answers is not in the document]
Comment (siqbal1986) [reply tests/vxlan/test_vxlan_route_advertisement.py]: @StormLiangMS  I have made the changes, now the test is looking at the T2s.

[the remark this answers is not in the document]
Comment (siqbal1986) [reply tests/vxlan/test_vxlan_route_advertisement.py]: @StormLiangMS  please review and approve so that this can be merged

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Pull request merged**